### PR TITLE
Add SPN status support for notifier

### PR DESCRIPTION
**Pull request by dhaavi**
- Refactor notifier icons, copy to go source tree for embedding
- Add SPN status support
- Switch to new embedded icon
- Update deps

### CODE REVIEW

**Comment by northys** [comment]
LOL, those icons are ready even for 16K resolution :D

File: notifier/icons.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import (
+	"io/ioutil"
+	"path/filepath"
+	"sync"
+
+	"github.com/safing/portmaster-ui/notifier/icons"
+)
+
+var (
+	appIconEnsureOnce = new(sync.Once)
```
Comment (ppacher) [inline notifier/icons.go:12]: There's no need to use `new()` here. The zero value of `sync.Once` is useable. Just don't copy it

File: notifier/icons.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	appIconPath       string
+)
+
+func ensureAppIcon() (location string, err error) {
+	appIconEnsureOnce.Do(func() {
+		if appIconPath == "" {
+			appIconPath = filepath.Join(dataDir, "exec", "portmaster.png")
+		}
+		err = ioutil.WriteFile(appIconPath, icons.PNG, 0o0644)
```
Comment (ppacher) [inline notifier/icons.go:21]: you actually state that portmaster processes should never read from the `exec` dir ....

**Review by ppacher — APPROVED** [review]

File: notifier/icons.go (hunk 1 above)
Comment (dhaavi) [reply notifier/icons.go:12]: Ah, yes. I planned to reset sync.Once when it fails, but then decided not to.

File: notifier/icons.go (hunk 2 above)
Comment (dhaavi) [reply notifier/icons.go:21]: Discussed: Due to lack of better (and quick) alternatives, we will stick with this for now.
Long-term the Portmaster could maintain icons files in the main directory.

**Pull request merged**